test/verify/test_nonstd_gather.cpp

```diff
@@ -0,0 +1,26 @@
+
+#include "verify_program.hpp"
+#include <migraphx/program.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/make_op.hpp>
+
+struct test_nonstd_gather : verify_program<test_nonstd_gather>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape s{migraphx::shape::float_type, {3, 3}};
+        migraphx::shape s_indices{migraphx::shape::int32_type, {2, 2}};
+        std::vector<int> indices{1, 1, 0, 2};
+        auto d  = mm->add_parameter("data", s);
+        auto td = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {1, 0}}}), d);
```

Comment:
Have you checked if simplify_reshapes is not removing two transposes ? since they generate a `no_op`.

Reply:
Even if they are not removed, `auto_contiguous` will add contiguous for both transposes that would make input to gather a standard shape. 

Because, after insertion of `contigous` in `auto_contiguous` shapes are recomputed for all its ouputs. that would make output of second transpose non-standard and `auto_contigous` will add contigous for it. So gather would receive standard shape.